File: soft uni prgramming fundamentals/6. Lists/Lists/src/ArrayManipulator.java
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class ArrayManipulator {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        List<Integer> list = Arrays.stream(s.nextLine().split(" "))
                .map(Integer::parseInt)
                .collect(Collectors.toList());
        String[] comands = s.nextLine().split(" ");


        while (!comands[0].equals("print")) {
            switch (comands[0]) {
                case "add":
                    int index = Integer.parseInt(comands[1]);
                    int element = Integer.parseInt(comands[2]);
                    list.add(index, element);
                    break;

                case "addMany":
                    int indexM = Integer.parseInt(comands[1]);
                    int count = 0;
                    for (int i = 2; i < comands.length; i++) {
                        list.add(indexM + count, Integer.parseInt(comands[i]));
                        count++;
                    }
                    break;
                case "contains":
                    int searchElement = Integer.parseInt(comands[1]);
                    boolean isFound = true;
                    for (int i = 0; i < list.size(); i++) {
                        if (list.get(i).equals(searchElement)) {
                            System.out.println(i);
                            isFound = true;
                            break;
                        }
                        isFound = false;
                    }
                    if (!isFound) {
                        System.out.println("-1");
                    }
                    break;
                case "remove":
                    int removeIndex = Integer.parseInt(comands[1]);
                    list.remove(removeIndex);
                    break;
                case "shift":
                    int n = Integer.parseInt(comands[1]);
                    n = n % list.size();
                    List<Integer> helper = list.subList(n, list.size());
                    for (int i = 0; i < n; i++) {
                        helper.add(list.get(i));
                    }
                    list = helper;
                    break;
                case "sumPairs":
                    int lenght = list.size();
                    for (int i = 0; i < lenght / 2; i++) {
                        list.set(i, list.get(i) + list.get(i + 1));
                        list.remove(i + 1);
                    }
                    break;
                default:
                    break;
            }
            comands = s.nextLine().split(" ");
        }


        if (comands[0].equals("print")) {
            System.out.print("[");
            for (int i = 0; i < list.size() - 1; i++) {
                System.out.print(list.get(i) + ", ");
            }
            System.out.println(list.get(list.size() - 1) + "]");
        }

    }
}

File: soft uni prgramming fundamentals/6. Lists/Lists C#/Lists/ArrayManipulator/ArrayManipulator.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ArrayManipulator
{
    class ArrayManipulator
    {
        static void Main(string[] args)
        {
            List<int> inputNumbers = Console.ReadLine().Split().Select(int.Parse).ToList();

            string[] comands = Console.ReadLine().Split().ToArray();

            while (comands[0] != "print")
            {
                if (comands[0] == "add")
                {
                    int index = int.Parse(comands[1]);
                    int element = int.Parse(comands[2]);

                    inputNumbers.Insert(index, element);
                }
                else if (comands[0] == "addMany")
                {
                    int index = int.Parse(comands[1]);
                    inputNumbers.InsertRange(index, comands.Skip(2).Select(int.Parse));
                }
                else if (comands[0] == "contains")
                {
                    int element = int.Parse(comands[1]);
                    if (inputNumbers.Contains(element))
                    {
                        Console.WriteLine(inputNumbers.IndexOf(element));
                    }
                    else
                    {
                        Console.WriteLine(-1);
                    }
                }
                else if (comands[0] == "remove")
                {
                    int index = int.Parse(comands[1]);
                    inputNumbers.RemoveAt(index);
                }
                else if (comands[0] == "shift")
                {
                    int position = int.Parse(comands[1])%inputNumbers.Count();

                    List<int> helper = inputNumbers.Skip(position).ToList();
                    
                    for (int i = 0; i < position; i++)
                    {
                        helper.Add(inputNumbers[i]);
                    }
                    inputNumbers = helper;
                }
                else if (comands[0] == "sumPairs")
                {
                    int lenght = inputNumbers.Count()/2;
                   
                    for (int i = 0; i < lenght ; i++)
                    {
                        inputNumbers[i]+=inputNumbers[i + 1];
                        inputNumbers.RemoveAt(i + 1);
                        
                    }
                    
                }
                comands = Console.ReadLine().Split().ToArray();
            }

            if (comands[0] == "print")
            {
                Console.WriteLine("[" + string.Join(", ", inputNumbers) + "]");
            }
        }
    }
}
